source/common/config/version_converter.cc

```diff
@@ -40,5 +92,59 @@ DowngradedMessagePtr VersionConverter::downgrade(const Protobuf::Message& messag
   return downgraded_message;
 }
 
+DynamicMessagePtr
+VersionConverter::reinterpret(const Protobuf::Message& message,
+                              envoy::config::core::v3alpha::ApiVersion api_version) {
+  switch (api_version) {
+  case envoy::config::core::v3alpha::ApiVersion::AUTO:
+  case envoy::config::core::v3alpha::ApiVersion::V2:
+    // TODO(htuch): this works as long as there are no new fields in the v3+
+    // DiscoveryRequest. When they are added, we need to do a full v2 conversion
+    // and also discard unknown fields. Tracked at
+    // https://github.com/envoyproxy/envoy/issues/9619.
+    return downgrade(message);
+  case envoy::config::core::v3alpha::ApiVersion::V3ALPHA: {
```

Comment:
Are we planning to support V3 over REST?

Reply:
Yes